### ISSUE TITLE
Unable to install "Purchase Management" add-on in fresh install

### ISSUE THREAD

**Issue by TheDailySpank** [bitnami/bitnami-docker-odoo#57]
**Description**

Attempt to install Purchase Management app fails with an error

**Steps to reproduce the issue:**

1. Download docker-compose.yml from repository
2. Run ```docker-compose up -d```
3. Login to Odoo
4. Go to Apps
5. Click "Install" button on "Purchase Management" App

**Describe the results you received:**

```
Error:
Odoo Server Error

Traceback (most recent call last):
  File "/opt/bitnami/odoo/odoo/models.py", line 1037, in _validate_fields
    check(self)
  File "/opt/bitnami/odoo/odoo/addons/base/ir/ir_ui_view.py", line 342, in _check_xml
    self.postprocess_and_fields(view.model, view_doc, view.id)
  File "/opt/bitnami/odoo/odoo/addons/base/ir/ir_ui_view.py", line 1077, in postprocess_and_fields
    self.raise_view_error("\n".join(msg_lines), view_id)
  File "/opt/bitnami/odoo/odoo/addons/base/ir/ir_ui_view.py", line 519, in raise_view_error
    raise ValueError(message)
ValueError: Field 'company_id' used in attributes must be present in view but is missing:
 - 'company_id' in domain="['|', ('company_id', '=', company_id), ('company_id', '=', False)]"
 - 'company_id' in domain="['|', ('company_id', '=', company_id), ('company_id', '=', False)]"

Error context:
View `Operation Types`
[view_id: 601, xml_id: n/a, model: stock.picking.type, parent_id: n/a]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/bitnami/odoo/odoo/tools/convert.py", line 744, in parse
    self._tags[rec.tag](rec, de, mode=mode)
  File "/opt/bitnami/odoo/odoo/tools/convert.py", line 654, in _tag_record
    id = self.env(context=rec_context)['ir.model.data']._update(rec_model, self.module, res, rec_id or False, not self.isnoupdate(data_node), noupdate=self.isnoupdate(data_node), mode=self.mode)
  File "/opt/bitnami/odoo/odoo/addons/base/ir/ir_model.py", line 1493, in _update
    record = record.create(values)
  File "/opt/bitnami/odoo/odoo/addons/base/ir/ir_ui_view.py", line 408, in create
    return super(View, self).create(self._compute_defaults(values))
  File "/opt/bitnami/odoo/odoo/models.py", line 3402, in create
    self._fields[key].determine_inverse(record)
  File "/opt/bitnami/odoo/odoo/fields.py", line 1073, in determine_inverse
    getattr(records, self.inverse)()
  File "/opt/bitnami/odoo/odoo/addons/base/ir/ir_ui_view.py", line 273, in _inverse_arch
    view.write(data)
  File "/opt/bitnami/odoo/odoo/addons/base/ir/ir_ui_view.py", line 424, in write
    return super(View, self).write(self._compute_defaults(vals))
  File "/opt/bitnami/odoo/odoo/models.py", line 3109, in write
    self._write(old_vals)
  File "/opt/bitnami/odoo/odoo/models.py", line 3277, in _write
    self._validate_fields(vals)
  File "/opt/bitnami/odoo/odoo/models.py", line 1041, in _validate_fields
    raise ValidationError("%s\n\n%s" % (_("Error while validating constraint"), tools.ustr(e)))
odoo.exceptions.ValidationError: ('Error while validating constraint\n\nField \'company_id\' used in attributes must be present in view but is missing:\n - \'company_id\' in domain="[\'|\', (\'company_id\', \'=\', company_id), (\'company_id\', \'=\', False)]"\n - \'company_id\' in domain="[\'|\', (\'company_id\', \'=\', company_id), (\'company_id\', \'=\', False)]"\n\nError context:\nView `Operation Types`\n[view_id: 601, xml_id: n/a, model: stock.picking.type, parent_id: n/a]', None)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/bitnami/odoo/odoo/http.py", line 651, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/opt/bitnami/odoo/odoo/http.py", line 310, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/opt/bitnami/odoo/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/opt/bitnami/odoo/odoo/http.py", line 693, in dispatch
    result = self._call_function(**self.params)
  File "/opt/bitnami/odoo/odoo/http.py", line 342, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/opt/bitnami/odoo/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/opt/bitnami/odoo/odoo/http.py", line 335, in checked_call
    result = self.endpoint(*a, **kw)
  File "/opt/bitnami/odoo/odoo/http.py", line 937, in __call__
    return self.method(*args, **kw)
  File "/opt/bitnami/odoo/odoo/http.py", line 515, in response_wrap
    response = f(*args, **kw)
  File "/opt/bitnami/odoo/odoo/addons/web/controllers/main.py", line 938, in call_button
    action = self._call_kw(model, method, args, {})
  File "/opt/bitnami/odoo/odoo/addons/web/controllers/main.py", line 926, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/opt/bitnami/odoo/odoo/api.py", line 689, in call_kw
    return call_kw_multi(method, model, args, kwargs)
  File "/opt/bitnami/odoo/odoo/api.py", line 680, in call_kw_multi
    result = method(recs, *args, **kwargs)
  File "<decorator-gen-42>", line 2, in button_immediate_install
  File "/opt/bitnami/odoo/odoo/addons/base/module/module.py", line 71, in check_and_log
    return method(self, *args, **kwargs)
  File "/opt/bitnami/odoo/odoo/addons/base/module/module.py", line 449, in button_immediate_install
    return self._button_immediate_function(type(self).button_install)
  File "/opt/bitnami/odoo/odoo/addons/base/module/module.py", line 543, in _button_immediate_function
    modules.registry.Registry.new(self._cr.dbname, update_module=True)
  File "/opt/bitnami/odoo/odoo/modules/registry.py", line 85, in new
    odoo.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/opt/bitnami/odoo/odoo/modules/loading.py", line 380, in load_modules
    loaded_modules, update_module, models_to_check)
  File "/opt/bitnami/odoo/odoo/modules/loading.py", line 274, in load_marked_modules
    perform_checks=perform_checks, models_to_check=models_to_check
  File "/opt/bitnami/odoo/odoo/modules/loading.py", line 181, in load_module_graph
    _load_data(cr, module_name, idref, mode, kind='data')
  File "/opt/bitnami/odoo/odoo/modules/loading.py", line 95, in _load_data
    tools.convert_file(cr, module_name, filename, idref, mode, noupdate, kind, report)
  File "/opt/bitnami/odoo/odoo/tools/convert.py", line 791, in convert_file
    convert_xml_import(cr, module, fp, idref, mode, noupdate, report)
  File "/opt/bitnami/odoo/odoo/tools/convert.py", line 852, in convert_xml_import
    obj.parse(doc.getroot(), mode=mode)
  File "/opt/bitnami/odoo/odoo/tools/convert.py", line 751, in parse
    exc_info[2]
  File "/opt/bitnami/odoo/odoo/tools/pycompat.py", line 86, in reraise
    raise value.with_traceback(tb)
  File "/opt/bitnami/odoo/odoo/tools/convert.py", line 744, in parse
    self._tags[rec.tag](rec, de, mode=mode)
  File "/opt/bitnami/odoo/odoo/tools/convert.py", line 654, in _tag_record
    id = self.env(context=rec_context)['ir.model.data']._update(rec_model, self.module, res, rec_id or False, not self.isnoupdate(data_node), noupdate=self.isnoupdate(data_node), mode=self.mode)
  File "/opt/bitnami/odoo/odoo/addons/base/ir/ir_model.py", line 1493, in _update
    record = record.create(values)
  File "/opt/bitnami/odoo/odoo/addons/base/ir/ir_ui_view.py", line 408, in create
    return super(View, self).create(self._compute_defaults(values))
  File "/opt/bitnami/odoo/odoo/models.py", line 3402, in create
    self._fields[key].determine_inverse(record)
  File "/opt/bitnami/odoo/odoo/fields.py", line 1073, in determine_inverse
    getattr(records, self.inverse)()
  File "/opt/bitnami/odoo/odoo/addons/base/ir/ir_ui_view.py", line 273, in _inverse_arch
    view.write(data)
  File "/opt/bitnami/odoo/odoo/addons/base/ir/ir_ui_view.py", line 424, in write
    return super(View, self).write(self._compute_defaults(vals))
  File "/opt/bitnami/odoo/odoo/models.py", line 3109, in write
    self._write(old_vals)
  File "/opt/bitnami/odoo/odoo/models.py", line 3277, in _write
    self._validate_fields(vals)
  File "/opt/bitnami/odoo/odoo/models.py", line 1041, in _validate_fields
    raise ValidationError("%s\n\n%s" % (_("Error while validating constraint"), tools.ustr(e)))
odoo.tools.convert.ParseError: "Error while validating constraint

Field 'company_id' used in attributes must be present in view but is missing:
 - 'company_id' in domain="['|', ('company_id', '=', company_id), ('company_id', '=', False)]"
 - 'company_id' in domain="['|', ('company_id', '=', company_id), ('company_id', '=', False)]"

Error context:
View `Operation Types`
[view_id: 601, xml_id: n/a, model: stock.picking.type, parent_id: n/a]
None" while parsing /opt/bitnami/odoo/odoo/addons/stock/views/stock_picking_views.xml:28, near
<record model="ir.ui.view" id="view_picking_type_form">
            <field name="name">Operation Types</field>
            <field name="model">stock.picking.type</field>
            <field name="arch" type="xml">
                <form string="Operation Types">
                    <sheet>
                        <div class="oe_button_box" name="button_box">
                            <button name="toggle_active" type="object" class="oe_stat_button" icon="fa-archive">
                                <field name="active" widget="boolean_button" options="{&quot;terminology&quot;: &quot;archive&quot;}"/>
                            </button>
                        </div>
                        <group>
                            <group>
                                <field name="name"/>
                                <field name="sequence_id"/>
                                <field name="warehouse_id" groups="stock.group_stock_multi_warehouses"/>
                            </group>
                            <group>
                                <field name="code"/>
                                <field attrs="{&quot;invisible&quot;: [(&quot;code&quot;, &quot;not in&quot;, [&quot;incoming&quot;, &quot;outgoing&quot;, &quot;internal&quot;])]}" name="return_picking_type_id"/>
                                <field attrs="{&quot;invisible&quot;: [(&quot;code&quot;, &quot;not in&quot;, [&quot;incoming&quot;, &quot;outgoing&quot;, &quot;internal&quot;])]}" name="barcode_nomenclature_id" groups="base.group_no_one"/>
                                <field name="show_operations"/>
                                <field name="show_reserved" attrs="{'invisible': [('show_operations', '!=', False)]}"/>
                            </group>
                        </group>
                        <group>
                            <group attrs="{&quot;invisible&quot;: [(&quot;code&quot;, &quot;not in&quot;, [&quot;incoming&quot;, &quot;outgoing&quot;, &quot;internal&quot;])]}" string="Packs and Lots" groups="stock.group_tracking_lot,stock.group_production_lot">
                                <field name="show_entire_packs" groups="stock.group_tracking_lot"/>
                                <field name="use_create_lots" groups="stock.group_production_lot"/>
                                <field name="use_existing_lots" groups="stock.group_production_lot"/>
                            </group>
                            <group string="Locations" groups="stock.group_stock_multi_locations">
                                <field name="default_location_src_id" attrs="{'required': [('code', '=', 'internal')]}" domain="['|', ('company_id', '=', company_id), ('company_id', '=', False)]"/>
                                <field name="default_location_dest_id" attrs="{'required': [('code', 'in', ('internal', 'incoming'))]}" domain="['|', ('company_id', '=', company_id), ('company_id', '=', False)]"/>
                            </group>
                        </group>
                    </sheet>
                </form>
            </field>
        </record>
```

**Describe the results you expected:**

Expected the App to install like all other Apps.

**Additional information you deem important (e.g. issue happens only occasionally):**

This happens whenever the Purchase Management App is attempted to be installed, regardless of company info being changed, other modules installed before it, or even on a clean database with no demo data populated.

**Version**

- Output of `docker version`:

```
Client:
 Version:           18.09.3
 API version:       1.39
 Go version:        go1.10.8
 Git commit:        774a1f4
 Built:             Thu Feb 28 06:40:58 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.3
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.8
  Git commit:       774a1f4
  Built:            Thu Feb 28 05:59:55 2019
  OS/Arch:          linux/amd64
  Experimental:     false

```

- Output of `docker info`:

```
Containers: 2
 Running: 2
 Paused: 0
 Stopped: 0
Images: 19
Server Version: 18.09.3
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 41
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: e6b3f5632f50dbc4e9cb6288d911bf4f5e95b18e
runc version: 6635b4f0c6af3810594d2770f662f34ddc15b40d
init version: fec3683
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.4.0-141-generic
Operating System: Ubuntu 16.04.5 LTS
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 3.859GiB
Name: MarvinFife
ID: OZVZ:J7IM:T36F:SWOF:MLLG:5BUJ:GGWO:ENQI:ERCK:2XZZ:S3RZ:7DG3
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Labels:
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
Product License: Community Engine

WARNING: No swap limit support

```

- Output of `docker-compose version` (if applicable):

```
docker-compose version 1.18.0, build 8dd22a9
docker-py version: 2.6.1
CPython version: 2.7.13
OpenSSL version: OpenSSL 1.0.1t  3 May 2016
```

- Odoo version info from inside app (under Settings)

```
Odoo 11.0-20190215 (Community Edition)
```

**Additional environment details (AWS, VirtualBox, Docker for MAC, physical, etc.):**

Tested on:
* Docker on Ubuntu 16 on Digital Ocean
* Docker for Mac

Bitnami VirtualBox version works as expected. Odoo 11.0-20190118 (Community Edition)

Note: Working VBox version is from 1/18 and non-working Docker version is 2/15

Might be Odoo version dependent, and if so, another version of it might need to be referenced by the Bitnami docker-compose.yml file.

**Comment by TheDailySpank** [comment]
**Update**

As a workaround, changing the version to an older version resolved the issue

docker-compose.yml
```
 odoo:
    image: 'bitnami/odoo:11.0.20190115'
```

**Comment by tompizmor** [comment]
Hello @TheDailySpank,

Just a quick note to let you know that I was able to reproduce the issue. We will take a deeper look to the error and update this issue once we have more information.

I am glad that you found a working version in the meantime.

**Comment by mumrau** [comment]
For your information, we have a similar issue with "Inventory Management" (a very similar one, `company_id` being the root of the problem for sure).

I'm giving a shot to downgrading like @TheDailySpank, I'll edit my post accordingly.

**Comment by tompizmor** [comment]
@TheDailySpank @mumrau  I tried again both modules, "Inventory Management" and "Purchase Management" in the latest version and they installed properly. 
Probably it was a bug introduced in some specific versions of Odoo. Can you try using the latest version of the Bitnami Odoo container?

**Comment by mumrau** [comment]
Nevermind, on a completely fresh install, without demo data, on `11.0.20190215`, I cannot get the Inventory Management applicaiont:
```
Error:
Odoo Server Error

Traceback (most recent call last):
  File "/opt/bitnami/odoo/odoo/models.py", line 1037, in _validate_fields
    check(self)
  File "/opt/bitnami/odoo/odoo/addons/base/ir/ir_ui_view.py", line 342, in _check_xml
    self.postprocess_and_fields(view.model, view_doc, view.id)
  File "/opt/bitnami/odoo/odoo/addons/base/ir/ir_ui_view.py", line 1077, in postprocess_and_fields
    self.raise_view_error("\n".join(msg_lines), view_id)
  File "/opt/bitnami/odoo/odoo/addons/base/ir/ir_ui_view.py", line 519, in raise_view_error
    raise ValueError(message)
ValueError: Field 'company_id' used in attributes must be present in view but is missing:
 - 'company_id' in domain="['|', ('company_id', '=', company_id), ('company_id', '=', False)]"
 - 'company_id' in domain="['|', ('company_id', '=', company_id), ('company_id', '=', False)]"

Error context:
View `Operation Types`
[view_id: 445, xml_id: n/a, model: stock.picking.type, parent_id: n/a]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/bitnami/odoo/odoo/tools/convert.py", line 744, in parse
    self._tags[rec.tag](rec, de, mode=mode)
  File "/opt/bitnami/odoo/odoo/tools/convert.py", line 654, in _tag_record
    id = self.env(context=rec_context)['ir.model.data']._update(rec_model, self.module, res, rec_id or False, not self.isnoupdate(data_node), noupdate=self.isnoupdate(data_node), mode=self.mode)
  File "/opt/bitnami/odoo/odoo/addons/base/ir/ir_model.py", line 1493, in _update
    record = record.create(values)
  File "/opt/bitnami/odoo/odoo/addons/base/ir/ir_ui_view.py", line 408, in create
    return super(View, self).create(self._compute_defaults(values))
  File "/opt/bitnami/odoo/odoo/models.py", line 3402, in create
    self._fields[key].determine_inverse(record)
  File "/opt/bitnami/odoo/odoo/fields.py", line 1073, in determine_inverse
    getattr(records, self.inverse)()
  File "/opt/bitnami/odoo/odoo/addons/base/ir/ir_ui_view.py", line 273, in _inverse_arch
    view.write(data)
  File "/opt/bitnami/odoo/odoo/addons/base/ir/ir_ui_view.py", line 424, in write
    return super(View, self).write(self._compute_defaults(vals))
  File "/opt/bitnami/odoo/odoo/models.py", line 3109, in write
    self._write(old_vals)
  File "/opt/bitnami/odoo/odoo/models.py", line 3277, in _write
    self._validate_fields(vals)
  File "/opt/bitnami/odoo/odoo/models.py", line 1041, in _validate_fields
    raise ValidationError("%s\n\n%s" % (_("Error while validating constraint"), tools.ustr(e)))
odoo.exceptions.ValidationError: ('Error while validating constraint\n\nField \'company_id\' used in attributes must be present in view but is missing:\n - \'company_id\' in domain="[\'|\', (\'company_id\', \'=\', company_id), (\'company_id\', \'=\', False)]"\n - \'company_id\' in domain="[\'|\', (\'company_id\', \'=\', company_id), (\'company_id\', \'=\', False)]"\n\nError context:\nView `Operation Types`\n[view_id: 445, xml_id: n/a, model: stock.picking.type, parent_id: n/a]', None)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/bitnami/odoo/odoo/http.py", line 651, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/opt/bitnami/odoo/odoo/http.py", line 310, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/opt/bitnami/odoo/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/opt/bitnami/odoo/odoo/http.py", line 693, in dispatch
    result = self._call_function(**self.params)
  File "/opt/bitnami/odoo/odoo/http.py", line 342, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/opt/bitnami/odoo/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/opt/bitnami/odoo/odoo/http.py", line 335, in checked_call
    result = self.endpoint(*a, **kw)
  File "/opt/bitnami/odoo/odoo/http.py", line 937, in __call__
    return self.method(*args, **kw)
  File "/opt/bitnami/odoo/odoo/http.py", line 515, in response_wrap
    response = f(*args, **kw)
  File "/opt/bitnami/odoo/odoo/addons/web/controllers/main.py", line 938, in call_button
    action = self._call_kw(model, method, args, {})
  File "/opt/bitnami/odoo/odoo/addons/web/controllers/main.py", line 926, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/opt/bitnami/odoo/odoo/api.py", line 689, in call_kw
    return call_kw_multi(method, model, args, kwargs)
  File "/opt/bitnami/odoo/odoo/api.py", line 680, in call_kw_multi
    result = method(recs, *args, **kwargs)
  File "<decorator-gen-42>", line 2, in button_immediate_install
  File "/opt/bitnami/odoo/odoo/addons/base/module/module.py", line 71, in check_and_log
    return method(self, *args, **kwargs)
  File "/opt/bitnami/odoo/odoo/addons/base/module/module.py", line 449, in button_immediate_install
    return self._button_immediate_function(type(self).button_install)
  File "/opt/bitnami/odoo/odoo/addons/base/module/module.py", line 543, in _button_immediate_function
    modules.registry.Registry.new(self._cr.dbname, update_module=True)
  File "/opt/bitnami/odoo/odoo/modules/registry.py", line 85, in new
    odoo.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/opt/bitnami/odoo/odoo/modules/loading.py", line 380, in load_modules
    loaded_modules, update_module, models_to_check)
  File "/opt/bitnami/odoo/odoo/modules/loading.py", line 274, in load_marked_modules
    perform_checks=perform_checks, models_to_check=models_to_check
  File "/opt/bitnami/odoo/odoo/modules/loading.py", line 181, in load_module_graph
    _load_data(cr, module_name, idref, mode, kind='data')
  File "/opt/bitnami/odoo/odoo/modules/loading.py", line 95, in _load_data
    tools.convert_file(cr, module_name, filename, idref, mode, noupdate, kind, report)
  File "/opt/bitnami/odoo/odoo/tools/convert.py", line 791, in convert_file
    convert_xml_import(cr, module, fp, idref, mode, noupdate, report)
  File "/opt/bitnami/odoo/odoo/tools/convert.py", line 852, in convert_xml_import
    obj.parse(doc.getroot(), mode=mode)
  File "/opt/bitnami/odoo/odoo/tools/convert.py", line 751, in parse
    exc_info[2]
  File "/opt/bitnami/odoo/odoo/tools/pycompat.py", line 86, in reraise
    raise value.with_traceback(tb)
  File "/opt/bitnami/odoo/odoo/tools/convert.py", line 744, in parse
    self._tags[rec.tag](rec, de, mode=mode)
  File "/opt/bitnami/odoo/odoo/tools/convert.py", line 654, in _tag_record
    id = self.env(context=rec_context)['ir.model.data']._update(rec_model, self.module, res, rec_id or False, not self.isnoupdate(data_node), noupdate=self.isnoupdate(data_node), mode=self.mode)
  File "/opt/bitnami/odoo/odoo/addons/base/ir/ir_model.py", line 1493, in _update
    record = record.create(values)
  File "/opt/bitnami/odoo/odoo/addons/base/ir/ir_ui_view.py", line 408, in create
    return super(View, self).create(self._compute_defaults(values))
  File "/opt/bitnami/odoo/odoo/models.py", line 3402, in create
    self._fields[key].determine_inverse(record)
  File "/opt/bitnami/odoo/odoo/fields.py", line 1073, in determine_inverse
    getattr(records, self.inverse)()
  File "/opt/bitnami/odoo/odoo/addons/base/ir/ir_ui_view.py", line 273, in _inverse_arch
    view.write(data)
  File "/opt/bitnami/odoo/odoo/addons/base/ir/ir_ui_view.py", line 424, in write
    return super(View, self).write(self._compute_defaults(vals))
  File "/opt/bitnami/odoo/odoo/models.py", line 3109, in write
    self._write(old_vals)
  File "/opt/bitnami/odoo/odoo/models.py", line 3277, in _write
    self._validate_fields(vals)
  File "/opt/bitnami/odoo/odoo/models.py", line 1041, in _validate_fields
    raise ValidationError("%s\n\n%s" % (_("Error while validating constraint"), tools.ustr(e)))
odoo.tools.convert.ParseError: "Error while validating constraint

Field 'company_id' used in attributes must be present in view but is missing:
 - 'company_id' in domain="['|', ('company_id', '=', company_id), ('company_id', '=', False)]"
 - 'company_id' in domain="['|', ('company_id', '=', company_id), ('company_id', '=', False)]"

Error context:
View `Operation Types`
[view_id: 445, xml_id: n/a, model: stock.picking.type, parent_id: n/a]
None" while parsing /opt/bitnami/odoo/odoo/addons/stock/views/stock_picking_views.xml:28, near
<record model="ir.ui.view" id="view_picking_type_form">
            <field name="name">Operation Types</field>
            <field name="model">stock.picking.type</field>
            <field name="arch" type="xml">
                <form string="Operation Types">
                    <sheet>
                        <div class="oe_button_box" name="button_box">
                            <button name="toggle_active" type="object" class="oe_stat_button" icon="fa-archive">
                                <field name="active" widget="boolean_button" options="{&quot;terminology&quot;: &quot;archive&quot;}"/>
                            </button>
                        </div>
                        <group>
                            <group>
                                <field name="name"/>
                                <field name="sequence_id"/>
                                <field name="warehouse_id" groups="stock.group_stock_multi_warehouses"/>
                            </group>
                            <group>
                                <field name="code"/>
                                <field attrs="{&quot;invisible&quot;: [(&quot;code&quot;, &quot;not in&quot;, [&quot;incoming&quot;, &quot;outgoing&quot;, &quot;internal&quot;])]}" name="return_picking_type_id"/>
                                <field attrs="{&quot;invisible&quot;: [(&quot;code&quot;, &quot;not in&quot;, [&quot;incoming&quot;, &quot;outgoing&quot;, &quot;internal&quot;])]}" name="barcode_nomenclature_id" groups="base.group_no_one"/>
                                <field name="show_operations"/>
                                <field name="show_reserved" attrs="{'invisible': [('show_operations', '!=', False)]}"/>
                            </group>
                        </group>
                        <group>
                            <group attrs="{&quot;invisible&quot;: [(&quot;code&quot;, &quot;not in&quot;, [&quot;incoming&quot;, &quot;outgoing&quot;, &quot;internal&quot;])]}" string="Packs and Lots" groups="stock.group_tracking_lot,stock.group_production_lot">
                                <field name="show_entire_packs" groups="stock.group_tracking_lot"/>
                                <field name="use_create_lots" groups="stock.group_production_lot"/>
                                <field name="use_existing_lots" groups="stock.group_production_lot"/>
                            </group>
                            <group string="Locations" groups="stock.group_stock_multi_locations">
                                <field name="default_location_src_id" attrs="{'required': [('code', '=', 'internal')]}" domain="['|', ('company_id', '=', company_id), ('company_id', '=', False)]"/>
                                <field name="default_location_dest_id" attrs="{'required': [('code', 'in', ('internal', 'incoming'))]}" domain="['|', ('company_id', '=', company_id), ('company_id', '=', False)]"/>
                            </group>
                        </group>
                    </sheet>
                </form>
            </field>
        </record>
```
I'm rollbacking again.

**Comment by tompizmor** [comment]
You are using `11.0.20190215` which is not the latest version. I can also reproduce the error on that specific version, however, it is working fine in the latest tag `11.0.20190315` (Note they are different). Have you tried with this tag?

**Comment by mumrau** [comment]
I'll give it a shot, this version is very new I didn't see the latest release.

I'm trying to solve issues related to #37 as well...
I'll come back to this thread whenever I get something completely working (or completely failed as well :p ).

**Comment by tompizmor** [comment]
The #37 issue is probably because Odoo initialization is failing at some point after creating the database user. Later on, Kubernetes restarts the pod and it will try to create another database user with the same name. 

I would try to get the logs of the first failing pod to see the real issue.

**Comment by mumrau** [comment]
Okay so I confirm that it works on a fresh install if you're not using the demo data.
It basically proves that the `demo mode` is missing some data for some addons installation.

Anyway, that's not really a problem, but I do believe that some documentation stating that demonstration mode is a thing, cause I had no idea about that. It could be an interesting evolution to the docs, and could be pretty helpful.

**Comment by tompizmor** [comment]
Thanks for the feedback @mumrau!
We will take into account and improve the documentation.

**Comment by marcosbc** [comment]
Hi @TheDailySpank, just wanted to let you know that we released `12.0.20210615-debian-10-r20`/`13.0.20210610-debian-10-r24`/`14.0.20210610-debian-10-r22` versions of this container which includes a major refactor of the existing logic. Refer to the [Notable Changes section in the README](https://github.com/bitnami/bitnami-docker-odoo#notable-changes) for more info.

As or this specific issue, I tried installing the "Purchase" addon in Odoo 14 and it worked perfectly. So I'm considering this issue to be solved. Let us know if you are continuing to find issues.
